### DISCUSSION TITLE
`Library not loaded: '@rpath/Frameworks/Python.framework/Versions/3.9/Python'`

### DISCUSSION THREAD

**Discussion by Kaiyangshi-Ito** [dashingsoft/pyarmor#1153] [Q&A] [answered]
I tried `pyarmor reg pyarmor-regfile-******.zip` with the latest version, upgraded successfully by `pip install pyarmor==8.1.3`, got:
```
INFO     Python 3.9.13
INFO     Pyarmor 8.1.1 (trial), 000000, non-profits
INFO     Platform darwin.aarch64
INFO     register "pyarmor-regfile-******.zip"
INFO     update license token
CRITICAL unknown error, please check pyarmor.error.log
ERROR    dlopen(/Users/my_username/anaconda3/lib/python3.9/site-packages/pyarmor/cli/core/pytransform3.so, 0x0002): Library not loaded: '@rpath/Frameworks/Python.framework/Versions/3.9/Python'
  Referenced from: '/Users/my_username/anaconda3/lib/python3.9/site-packages/pyarmor/cli/core/pytransform3.so'
  Reason: tried: '/Users/my_username/anaconda3/bin/../Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/Users/my_username/anaconda3/bin/../Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/Users/my_username/anaconda3/bin/../lib/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/Users/my_username/anaconda3/bin/../lib/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.9/Python' (no such file)
```
And it turns out that if I type `pyarmor -v`, it gives me:

```
Pyarmor 8.1.1 (trial), 000000, non-profits

License Type    : pyarmor-trial
License No.     : pyarmor-vax-000000
License To      : 

BCC Mode        : No
RFT Mode        : No

Notes
* Internet connection is required to verify Pyarmor license
* Trial license can't obfuscate big script and mix str
```

**Comment by jondy** [comment]
Could you find any `Python.framework` in the path `/Users/my_username/anaconda3/`?

If not, is there any `libpython*.so` or `libpython*.dylib` in the path `/Users/my_username/anaconda3/`?

**Comment by Kaiyangshi-Ito** [comment]
> 

no `Python.framework` in the path `/Users/my_username/anaconda3/`
no `libpython*.so` or `libpython*.dylib` either.

But under `/Users/my_username/anaconda3/lib/` there is `libpython3.9.dylib`

**Comment by jondy** [comment, marked as answer]
I document the solution here
https://pyarmor.readthedocs.io/en/latest/reference/errors.html

**Reply by Kaiyangshi-Ito** [reply]
Thanks for the solution. Can confirm now it works, but I recall that previously with Pyarmor 7, this was not needed. Is it possible to implement the same approach in previous versions?

**Comment by suss-userx** [comment]
do you have any suggestion for the following error

➜ pyarmor reg pyarmor-regfile-3500.zip
INFO     Python 3.9.6
INFO     Pyarmor 8.1.9 (trial), 000000, non-profits
INFO     Platform darwin.aarch64
INFO     register "pyarmor-regfile-3500.zip"
INFO     update license token
ERROR    dlopen(/Users/my_user_name/Library/Python/3.9/lib/python/site-packages/pyarmor/cli/core/pytransform3.so, 0x0002): Library not loaded: '@rpath/lib/libpython3.9.dylib'
  Referenced from: '/Users/my_user_name/Library/Python/3.9/lib/python/site-packages/pyarmor/cli/core/pytransform3.so'
  Reason: tried: '/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/../lib/libpython3.9.dylib' (no such file), '/Users/my_user_name/Library/Python/3.9/lib/python/site-packages/pyarmor/cli/core/../lib/libpython3.9.dylib' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib' (no such file), '/Library/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib' (no such file), '/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/../lib/libpython3.9.dylib' (no such file), '/Users/my_user_name/Library/Python/3.9/lib/python/site-packages/pyarmor/cli/core/../lib/libpython3.9.dylib' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib' (no such file), '/Library/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib' (no such file), '/usr/local/lib/libpython3.9.dylib' (no such file), '/usr/lib/libpython3.9.dylib' (no such file)
